src/Controller/AppController.php

```diff
@@ -378,15 +378,17 @@ protected function applySessionFilter(): ?Response
         }
 
         // write request query parameters (if any) in session
-        if (!empty($this->request->getQueryParams())) {
-            $session->write($sessionKey, $this->request->getQueryParams());
+        $params = $this->request->getQueryParams();
+        if (!empty($params)) {
+            unset($params['_search']);
+            $session->write($sessionKey, $params);
 
             return null;
         }
 
         // read request query parameters from session and redirect to proper page
         if ($session->check($sessionKey)) {
-            $query = http_build_query($session->read($sessionKey), null, '&', PHP_QUERY_RFC3986);
+            $query = http_build_query($session->read($sessionKey), '', '&', PHP_QUERY_RFC3986);
```

Comment:
To avoid infinite loops I suggest:

```php
        $params = (array)$session->read($sessionKey);
        if (!empty($params)) {
            $query = http_build_query($params, '', '&', PHP_QUERY_RFC3986);

            return $this->redirect(....);
        }
```